<a href="https://colab.research.google.com/github/cjhaster/NRSurrogateCatalog/blob/gh-pages/_sources/events/GW190805_211137.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open in Colab"/></a>

In [1]:
%load_ext autoreload
%autoreload 2
%matplotlib inline

import pandas as pd

pd.set_option("display.max_rows", None, "display.max_columns", None)

In [2]:
import pip

try:
    __import__("nrsur_catalog")
except ImportError:
    pip.main(["install", "nrsur_catalog", "-q", ])


# GW190805_211137

Below are some plots for GW190805_211137 from the NRSurrogate Catalog.

In [3]:
from nrsur_catalog import NRsurResult

nrsur_result = NRsurResult.load("GW190805_211137", cache_dir=".nrsur_catalog_cache")
# you can specify a `cache_dir`: folder where data will be downloaded

## Summary

In [4]:
nrsur_result.summary()

Posterior 90% CI  \
Parameter                                                         
$m_1\ [M_{\odot}]$                  ${86.35}_{-10.47}^{+13.06}$   
$m_2\ [M_{\odot}]$                  ${57.89}_{-14.43}^{+10.63}$   
$\mathcal{M}\ [M_{\odot}]$            ${60.52}_{-6.44}^{+5.24}$   
$q$                                    ${0.67}_{-0.22}^{+0.21}$   
$a_1$                                  ${0.75}_{-0.32}^{+0.17}$   
$a_2$                                  ${0.58}_{-0.37}^{+0.29}$   
$\theta_1$                             ${0.87}_{-0.41}^{+0.49}$   
$\theta_2$                             ${1.14}_{-0.58}^{+0.77}$   
$\chi_{\mathrm{eff}}$                  ${0.35}_{-0.21}^{+0.17}$   
$\chi_{\mathrm{p}}$                    ${0.53}_{-0.23}^{+0.23}$   
$\alpha$                               ${6.05}_{-5.49}^{+0.11}$   
$\delta$                              ${-0.18}_{-0.39}^{+0.37}$   
$t_c\ [s]$                    ${1249074715.31}_{-0.00}^{+0.01}$   
$d_L\ [Mpc]$                  ${6334.90}_{-2222.61}^{+2284.76}$   
$\phi$                                 ${3.20}_{-2.26}^{+2.19}$   
$\phi$                                 ${1.71}_{-0.27}^{+0.83}$   
$\theta$                               ${1.26}_{-0.06}^{+0.06}$   
$\psi$                                 ${1.54}_{-1.11}^{+1.15}$   
$\phi_{\mathrm{JL}}$                   ${2.96}_{-1.88}^{+2.18}$   
$\phi_{12}$                            ${3.14}_{-2.18}^{+2.18}$   
$\theta_{\mathrm{JN}}$                 ${0.88}_{-0.49}^{+1.64}$   
$m_f\ [M_{\odot}]$                   ${135.46}_{-9.55}^{+9.82}$   
$\chi_f$                               ${0.80}_{-0.08}^{+0.05}$   
$v_{\rm kick}\ [{\rm km/s}]$     ${894.00}_{-611.38}^{+906.77}$   

                                                                        Prior  
Parameter                                                                      
$m_1\ [M_{\odot}]$                                                        $-$  
$m_2\ [M_{\odot}]$                                                        $-$  
$\mathcal{M}\ [M_{\odot}]$    $\text{UniformInComponentsChirpMass} [12, 400]$  
$q$                           $\text{UniformInComponentsMassRatio} [0.17, 1]$  
$a_1$                                              $\text{Uniform} [0, 0.99]$  
$a_2$                                              $\text{Uniform} [0, 0.99]$  
$\theta_1$                                            $\text{Sine} [0, 3.14]$  
$\theta_2$                                            $\text{Sine} [0, 3.14]$  
$\chi_{\mathrm{eff}}$                                                     $-$  
$\chi_{\mathrm{p}}$                                                       $-$  
$\alpha$                                                                  $-$  
$\delta$                                                                  $-$  
$t_c\ [s]$                    $\text{Uniform} [1249074715.25, 1249074715.45]$  
$d_L\ [Mpc]$                     $\text{UniformSourceFrame} [100.0, 20000.0]$  
$\phi$                                             $\text{Uniform} [0, 6.28]$  
$\phi$                                             $\text{Uniform} [0, 6.28]$  
$\theta$                                              $\text{Sine} [0, 3.14]$  
$\psi$                                             $\text{Uniform} [0, 3.14]$  
$\phi_{\mathrm{JL}}$                               $\text{Uniform} [0, 6.28]$  
$\phi_{12}$                                        $\text{Uniform} [0, 6.28]$  
$\theta_{\mathrm{JN}}$                                $\text{Sine} [0, 3.14]$  
$m_f\ [M_{\odot}]$                                                        $-$  
$\chi_f$                                                                  $-$  
$v_{\rm kick}\ [{\rm km/s}]$                                              $-$



| Parameter                    | Posterior 90% CI                  | Prior                                           |
|:-----------------------------|:----------------------------------|:------------------------------------------------|
| $m_1\ [M_{\odot}]$           | ${86.35}_{-10.47}^{+13.06}$       | $-$                                             |
| $m_2\ [M_{\odot}]$           | ${57.89}_{-14.43}^{+10.63}$       | $-$                                             |
| $\mathcal{M}\ [M_{\odot}]$   | ${60.52}_{-6.44}^{+5.24}$         | $\text{UniformInComponentsChirpMass} [12, 400]$ |
| $q$                          | ${0.67}_{-0.22}^{+0.21}$          | $\text{UniformInComponentsMassRatio} [0.17, 1]$ |
| $a_1$                        | ${0.75}_{-0.32}^{+0.17}$          | $\text{Uniform} [0, 0.99]$                      |
| $a_2$                        | ${0.58}_{-0.37}^{+0.29}$          | $\text{Uniform} [0, 0.99]$                      |
| $\theta_1$                   | ${0.87}_{-0.41}^{+0.49}$          | $\text{Sine} [0, 3.14]$                         |
| $\theta_2$                   | ${1.14}_{-0.58}^{+0.77}$          | $\text{Sine} [0, 3.14]$                         |
| $\chi_{\mathrm{eff}}$        | ${0.35}_{-0.21}^{+0.17}$          | $-$                                             |
| $\chi_{\mathrm{p}}$          | ${0.53}_{-0.23}^{+0.23}$          | $-$                                             |
| $\alpha$                     | ${6.05}_{-5.49}^{+0.11}$          | $-$                                             |
| $\delta$                     | ${-0.18}_{-0.39}^{+0.37}$         | $-$                                             |
| $t_c\ [s]$                   | ${1249074715.31}_{-0.00}^{+0.01}$ | $\text{Uniform} [1249074715.25, 1249074715.45]$ |
| $d_L\ [Mpc]$                 | ${6334.90}_{-2222.61}^{+2284.76}$ | $\text{UniformSourceFrame} [100.0, 20000.0]$    |
| $\phi$                       | ${3.20}_{-2.26}^{+2.19}$          | $\text{Uniform} [0, 6.28]$                      |
| $\phi$                       | ${1.71}_{-0.27}^{+0.83}$          | $\text{Uniform} [0, 6.28]$                      |
| $\theta$                     | ${1.26}_{-0.06}^{+0.06}$          | $\text{Sine} [0, 3.14]$                         |
| $\psi$                       | ${1.54}_{-1.11}^{+1.15}$          | $\text{Uniform} [0, 3.14]$                      |
| $\phi_{\mathrm{JL}}$         | ${2.96}_{-1.88}^{+2.18}$          | $\text{Uniform} [0, 6.28]$                      |
| $\phi_{12}$                  | ${3.14}_{-2.18}^{+2.18}$          | $\text{Uniform} [0, 6.28]$                      |
| $\theta_{\mathrm{JN}}$       | ${0.88}_{-0.49}^{+1.64}$          | $\text{Sine} [0, 3.14]$                         |
| $m_f\ [M_{\odot}]$           | ${135.46}_{-9.55}^{+9.82}$        | $-$                                             |
| $\chi_f$                     | ${0.80}_{-0.08}^{+0.05}$          | $-$                                             |
| $v_{\rm kick}\ [{\rm km/s}]$ | ${894.00}_{-611.38}^{+906.77}$    | $-$                                             |


Lets make some plots!

In [5]:
# NRSurrogate corner plots

import os

param_sets = dict(
    mass=["mass_1", "mass_2", "chirp_mass", "mass_ratio"],
    spin=["a_1", "a_2", "tilt_1", "tilt_2"],
    effective_spin=["mass_ratio", "chi_eff", "chi_p"],
    sky_localisation=["luminosity_distance", "ra", "dec"],
    remnant=["final_mass", "final_spin", "final_kick"],
)
for name, params in param_sets.items():
    fname = f"GW190805_211137_{name}_corner.png"
    if not os.path.isfile(fname):
        fig = nrsur_result.plot_corner(params)
        fig.savefig(fname)

    if name == "remnant":
        continue

    # LVK-Comparison plots
    fname = f"GW190805_211137_compare_{name}_corner.png"
    if not os.path.isfile(fname):
        fig = nrsur_result.plot_lvk_comparison_corner(params)
        fig.savefig(fname)


## Corner Plots

### Mass


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW190805_211137_mass_corner.png"](GW190805_211137_mass_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW190805_211137_compare_mass_corner.png"](GW190805_211137_compare_mass_corner.png)
:::

::::



### Spin


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW190805_211137_spin_corner.png"](GW190805_211137_spin_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW190805_211137_compare_spin_corner.png"](GW190805_211137_compare_spin_corner.png)
:::

::::


### Effective Spin


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW190805_211137_effective_spin_corner.png"](GW190805_211137_effective_spin_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW190805_211137_compare_effective_spin_corner.png"](GW190805_211137_compare_effective_spin_corner.png)
:::

::::



### Sky-localisation


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW190805_211137_sky_localisation_corner.png"](GW190805_211137_sky_localisation_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW190805_211137_compare_sky_localisation_corner.png"](GW190805_211137_compare_sky_localisation_corner.png)
:::

::::



### Remnant

!["GW190805_211137_remnant_corner.png"](GW190805_211137_remnant_corner.png)


## Waveform posterior-predictive plot

In [6]:
fname = f"GW190805_211137_waveform.png"
if not os.path.isfile(fname):
    fig = nrsur_result.plot_signal(outdir=".")


![waveform](GW190805_211137_waveform.png)

## Animations  
Here are some animations of the posterior.

|Spin       | 
|-----------|
| <iframe width="630" height="315" src="https://www.youtube.com/embed/RWYEYlPOl14?autoplay=1&mute=1&playlist=RWYEYlPOl14&loop=1" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>    |   

|Remnant       | 
|-----------|
| <iframe width="630" height="315" src="https://www.youtube.com/embed/gZn4oRGkPp4?autoplay=1&mute=1&playlist=gZn4oRGkPp4&loop=1" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>    | 


## Analysis configs
Below are the configs used for the analysis of this job.

In [7]:
nrsur_result.print_configs()

Tmax:                               10000
accounting:                         None
accounting_user:                    None
adapt:                              False
allow_tape:                         True
analysis_executable:                None
analysis_executable_parser:         None
autocorr_c:                         5.0
autocorr_tol:                       50.0
bilby_zero_likelihood_mode:         False
burn_in_nact:                       50.0
calibration_model:                  CubicSpline
calibration_prior_boundary:         reflective
catch_waveform_errors:              True
channel_dict:                       {H1:DCS-CALIB_STRAIN_CLEAN_SUB60HZ_C01, L1:DCS-CALIB_STRAIN_CLEAN_SUB60HZ_C01, V1:Hrec_hoft_16384Hz}
check_point_deltaT:                 3600
clean:                              False
coherence_test:                     False
conda_env:                          None
condor_job_priority:                0
conversion_function:                None
create_summary:              

If you used this data, please [cite this work](../citation.md).

## Comments
Leave a comment below or in this [issue](https://github.com/avivajpeyi/nrsur_catlog/issues/new?title=GW190805_211137).
```{raw} html
<script src="https://utteranc.es/client.js"
        repo="avivajpeyi/nrsur_catalog"
        issue-term="GW190805_211137"
        theme="github-light"
        crossorigin="anonymous"
        async>
</script>
```